brayns/tasks/AddModelFromBlobTask.cpp

```diff
@@ -40,10 +40,11 @@ AddModelFromBlobTask::AddModelFromBlobTask(const BinaryParam& param,
 
     LoadModelFunctor functor{engine, param};
     functor.setCancelToken(_cancelToken);
-    functor.setProgressFunc([& progress = progress](const auto& msg, auto,
-                                                    auto amount) {
-        progress.update(msg, amount);
-    });
+    functor.setProgressFunc(
+        [& progress = progress, w = CHUNK_PROGRESS_WEIGHT ](const auto& msg,
+                                                            auto, auto amount) {
+            progress.update(msg, w + (amount * (1.f - w)));
```

Comment:
A minor comment. I'm starting to think that the progress callback should not be a single progress bar, but a bar per step and there should be a separate method to signal when the process is done instead of relying on reaching 1. With this code, if you upload a huge mesh to the server, transmission accounts for half the progress, creating the mesh is 1/4 and the remaining 1/4 is postprocessing. I don't have numbers but it post processing happens to be slower than anything else, the progress bar becomes a bit misleading.

Reply:
For more complicated loaders like for circuits, nesting of callbacks can or should be used today to mimic what you describe. The fixed amount for receiving the mesh can be tweaked for sure to account more the mesh loading/parsing itself, though if your link speed is slow, the upload takes more time. So it's not that easy to reflect that perfectly on the progress bar.